packages/core/src/animation/Animator.ts

```diff
@@ -0,0 +1,696 @@
+            playingStateData.frameTime %= playingStateData.state.clipEndTime;
+          } else {
+            playingStateData.frameTime = playingStateData.state.clipEndTime;
+            playingStateData.playType = PlayType.IsFinish;
+            ++finishLayerCount;
+          }
+        }
+      }
+      if (playingStateData.frameTime === PlayType.IsFinish) {
+        ++finishLayerCount;
+      }
+      this._updateLayer(i, isFirstLayer, deltaTime);
+    }
+    if (finishLayerCount === layerCount) {
+      for (let i = layerCount - 1; i >= 0; i--) {
+        const animatorLayerData = animatorLayersData[i];
+        const { playingStateData } = animatorLayerData;
+        this._revert(playingStateData);
+      }
+      this._playing = false;
+    }
+  }
+
+  /**
+   * Return the layer by name.
+   * @param name - The layer name
+   */
+  getLayerByName(name: string): AnimatorControllerLayer {
+    if (this.animatorController) {
+      return this.animatorController.findLayerByName(name);
+    }
+    return null;
+  }
+
+  /**
+   * @override
+   * @internal
+   */
+  _onEnable(): void {
+    this.engine._componentsManager.addOnUpdateAnimations(this);
+  }
+
+  /**
+   * @override
+   * @internal
+   */
+  _onDisable(): void {
+    this.engine._componentsManager.removeOnUpdateAnimations(this);
+  }
+
+  /**
+   * @internal
+   */
+  _setDefaultValueAndTarget(stateData: AnimatorStateData): void {
+    const { clip } = stateData.state;
+    if (clip) {
+      const curves = clip._curves;
+      for (let i = curves.length - 1; i >= 0; i--) {
+        const curve = curves[i];
+        const { relativePath, property } = curve;
+        const targetEntity = this.entity.findByPath(relativePath);
+        if (!stateData.curveDatas[i]) {
+          const curveData = new CurveData();
+          curveData.target = targetEntity;
+          switch (property) {
+            case AnimationProperty.Position:
+              curveData.defaultValue = targetEntity.transform.position.clone();
+              break;
+            case AnimationProperty.Rotation:
+              curveData.defaultValue = targetEntity.transform.rotationQuaternion.clone();
+              break;
+            case AnimationProperty.Scale:
+              curveData.defaultValue = targetEntity.transform.scale.clone();
+              break;
+          }
+          stateData.curveDatas[i] = curveData;
+        }
+      }
+    }
+  }
+
+  /**
+   * @internal
+   */
+  _setTempPoseValue(stateData: AnimatorStateData): void {
+    const { clip } = stateData.state;
+    if (clip) {
+      const curves = clip._curves;
+      for (let i = curves.length - 1; i >= 0; i--) {
+        const curve = curves[i];
+        const { property } = curve;
+        const curveData = stateData.curveDatas[i];
+        const targetEntity = curveData.target;
+        switch (property) {
+          case AnimationProperty.Position:
+            curveData.tempPoseValue = targetEntity.transform.position.clone();
```

Comment:
Shouldn't use `clone()` here, will cause GC!
You  should new tempPose when init,and use `cloneTo()` here!